test/metabase/api/setup_test.clj

```diff
@@ -272,6 +274,39 @@
         (with-setup {:database {:engine "h2", :name db-name}}
           (is (db/exists? Database :name db-name)))))))
 
+(deftest create-superuser-only-once-test
+  (testing "POST /api/setup"
+    (testing "Check that we cannot create a new superuser via setup-token when a user exists"
+      (let [token (setup/create-token!)
+            body  {:token token
+                   :prefs {:site_locale "es_MX"
+                           :site_name   (mt/random-name)}
+                   :user  {:first_name (mt/random-name)
+                           :last_name  (mt/random-name)
+                           :email      (mt/random-email)
+                           :password   "p@ssword1"}}]
+
+        (is (setup/has-user-setup))
+
+        (with-redefs [setup/has-user-setup (let [value (atom false)]
+                                             (fn
+                                               ([] @value)
+                                               ([x] (reset! value x))))]
```

Comment:
the setting has a `:setter :none`. Should this throw rather than reset the value? or is this just a helper to change the value in the test?

Reply:
I'll simplify this. As you said it doesn't need to be settable, all it needs to do is return a value when called with 0 args, and not accept 1 arg. Would it be worth using an actual setting with a :getter that returns false?

Reply:
I guess I'm wondering if this should be just `(binding [setup/has-user-setup (constantly false))`. Or if you want to change it after a few tests,

`(let [setup? (atom false)] (binding [setup/has-user-setup (fn [] @setup?)] (test) (reset! setup? true) (test asserting that you can do something now)`

Reply:
I thought about the `(constantly false)` piece a bit: I think it shouldn't be because I want an exception to occur if `setup/has-user-setup` arity-1 version gets called, because thats how the setting behaves with a `:none` `:getter`.

Reply:
The latest PR includes the switching to `true` of `setup/has-user-setup`.